### ISSUE TITLE
400 error results in checksum mismatch exception text

### ISSUE THREAD

**Issue by aaronlp** [BunnyWay/BunnyCDN.PHP.Storage#15]
When a 400 error is encountered from the API for file upload, the default/only exception message is `Bunny\Storage\Exception  Checksum and file contents mismatched.`

This is not a correct assumption to make in all cases. In my case, I had incorrectly attempted to upload a file without its extension and the API was actually returning the following error in the 400 response:

```
{"HttpCode":400,"Message":"Unable to upload file: Failed to create file, directory for path does not exist: /path/to/file"}
```

**Comment by aaronlp** [comment]
I've added a PR to address this issue: https://github.com/BunnyWay/BunnyCDN.PHP.Storage/pull/16

**Comment by rafael-at-bunny** [comment]
It's now merged and released in version 3.3.1.

Thank you for your contribution!
